prover/src/lib.rs

```diff
@@ -496,44 +480,53 @@ pub trait Prover {
         // - interpolate the trace into a polynomial in coefficient form
         // - "break" the polynomial into a set of column polynomials each of degree equal to
         //   trace_length - 1
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let composition_poly =
-            CompositionPoly::new(composition_poly_trace, domain, num_trace_poly_columns);
-        #[cfg(feature = "std")]
-        debug!(
-            "Converted constraint evaluations into {} composition polynomial columns of degree {} in {} ms",
-            composition_poly.num_columns(),
-            composition_poly.column_degree(),
-            now.elapsed().as_millis()
-        );
+        let composition_poly = info_span!(
+            "Converted constraint evaluations into composition polynomial columns",
+            columns_number = field::Empty,
+            degree = field::Empty
+        )
+        .in_scope(|| {
```

Comment:
Let's make the following changes here:

1. Change the name to "build_composition_poly_columns".
2. Rename `num_trace_poly_columns` parameter on line 473 above to `num_constraint_composition_columns`.
3. Let's use `num_columns` field and set it to `num_constraint_composition_columns`.
4. For `degree` field use `domain.trace_lenght() - 1`.
5. Assert that the values of `composition_poly.num_columns()` and `composition_poly.column_degree()` are as expected.